quora-service/src/main/java/com/upgrad/quora/service/dao/QuestionDao.java

```diff
@@ -26,4 +30,28 @@ public UserAuthEntity getUserAuthToken(final String accessToken) {
             return null;
         }
     }
+
+    public List<QuestionEntity> getAllQuestionsByUser(final String uuid){
+        try {
+            return entityManager.createNamedQuery("getAllQuestionsByUser", QuestionEntity.class).setParameter("uuid", uuid).getResultList();
+        } catch (NoResultException nre) {
+            return null;
+        }
+    }
+
+    public QuestionEntity getQuestionByUuid(final String uuid) {
+        try {
+            return entityManager.createNamedQuery("getQuestionByUuid", QuestionEntity.class).setParameter("uuid",uuid).getSingleResult();
+        } catch (NoResultException nre) {
+            return null;
+        }
+
+    }
+
+    public void userQuestionDelete(final String uuid) {
+
+        questionEntity = getQuestionByUuid(uuid);
+        entityManager.remove(questionEntity);
```

Comment:
Have you implemented `DELETE CASCADE`?

Reply:
No, because as per requirement only question needs to be deleted. However, if some deletes question then corresponding answer also should get deleted. what should we do?

Reply:
In given database schema `DELETE CASCADE` is already implemented so I think we don't need to worry about it.